### PR TITLE
2024 apriltags

### PR DESCRIPTION
**Pull request by lmaxwell24**
https://firstfrc.blob.core.windows.net/frc2024/FieldAssets/2024LayoutMarkingDiagram.pdf

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -4,14 +4,15 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-robotpy-pathplannerlib = "==2024.0.0b6.post3"
-phoenix6 = "==24.0.0b5"
```
Comment (LandonBayer) [inline Pipfile:8]: why no more phoenix version?

File: constants.py [hunk 2]
```diff
@@ -389,52 +389,100 @@
 
 kApriltagPositionDict = {  # thanks 6328 for FieldConstants!
```
Comment (LandonBayer) [inline constants.py:390]: i thought these are just in photonlib or smthing no? either way it's fine ig

File: Pipfile (hunk 1 above)
Comment (LandonBayer) [reply Pipfile:8]: It's in the lock away so ig it's fine

**Review by LandonBayer — APPROVED** [review]

File: Pipfile (hunk 1 above)
Comment (lmaxwell24) [reply Pipfile:8]: upgraded version

File: constants.py (hunk 2 above)
Comment (lmaxwell24) [reply constants.py:390]: not in photonlibpy (its lacking)

File: constants.py [hunk 3]
```diff
@@ -389,52 +389,100 @@
...
+        Rotation3d(0.0, 0.0, 120 * kRadiansPerDegree),
     ),
     3: Pose3d(
-        (kMetersPerInch * 610.77),
-        (kMetersPerInch * 174.19),  # FIRST's diagram has a typo (it says 147.19)
-        (kMetersPerInch * 18.22),
+        (kMetersPerInch * 652.73),
+        (kMetersPerInch * 196.17),
+        (kMetersPerInch * 57.13),
         Rotation3d(0.0, 0.0, math.pi),
```
Comment (icai0) [inline constants.py:407]: does 180 vs 0 make a difference?

Comment (icai0) [reply constants.py:407]: ignore this i forgot how to math

**Review by icai0 — APPROVED** [review]

**Pull request merged**